Header header
bool[] circuits_state